core/dictgen/src/rootcling_impl.cxx

```diff
@@ -2987,14 +2987,15 @@ class tempFileNamesCatalog {
 ////////////////////////////////////////////////////////////////////////////////
 /// Transform name of dictionary
 
-std::ostream *CreateStreamPtrForSplitDict(const std::string &dictpathname,
-      tempFileNamesCatalog &tmpCatalog)
+std::unique_ptr<std::ostream, void (*)(std::ostream *)>
+CreateStreamPtrForSplitDict(const std::string &dictpathname, tempFileNamesCatalog &tmpCatalog)
 {
    std::string splitDictName(tmpCatalog.getFileName(dictpathname));
    const size_t dotPos = splitDictName.find_last_of(".");
    splitDictName.insert(dotPos, "_classdef");
    tmpCatalog.addFileName(splitDictName);
-   return new std::ofstream(splitDictName.c_str());
+   return std::unique_ptr<std::ostream, void (*)(std::ostream *)>(new std::ofstream(splitDictName.c_str()),
```

Comment:
Why do we need a custom deleter for this unique_ptr? Can't we use `return llvm::make_unique<std::ostream>(splitDictName.c_str()));`?